workflows/catalog_upsert.go

```diff
@@ -275,6 +275,12 @@ func (workflow *catalogWorkflow) catalogProductVersions(namespace string, catalo
 					return err
 				}
 
+				if pipelineTemplate.Source.Provider == "S3" {
```

Comment:
this doesn't seem right.  the bucket/object shouldn't be provided until the pipeline is created, not when the service catalog product is created.